### PR TITLE
Allow full text rendering capabilities in layout text labels

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows for shadows, text buffers, etc in these labels

### CODE REVIEW

**Comment by nyalldawson** [comment]
Supersedes https://github.com/qgis/QGIS/pull/46269

File: src/core/layout/qgslayoutitemlabel.cpp [hunk 1]
```diff
@@ -424,7 +452,31 @@ bool QgsLayoutItemLabel::readPropertiesFromElement( const QDomElement &itemElem,
...
+    }
+    else if ( f.pixelSize() > 0 )
+    {
+      mFormat.setSize( f.pixelSize() );
+      mFormat.setSizeUnit( QgsUnitTypes::RenderPixels );
+    }
+  }
 
   //font color
   const QDomNodeList fontColorList = itemElem.elementsByTagName( QStringLiteral( "FontColor" ) );
```
Comment (nirvn) [inline src/core/layout/qgslayoutitemlabel.cpp:482]: @nyalldawson , I think could be (should be?) moved into the above else { // no text-style } block, logic would be easier to understand.

**Review by nirvn — APPROVED** [review]
The day has finally come! I love it.

**Comment by roya0045** [comment]
Thanks Nyall, this is going to be easier and cleaner with than the shoddy one I did. 😄

**Comment by Ronflex2k** [comment]
Hello, I'm very interested by this feature but I don't know how implement this thing in QGis. Do you have some advices ?

TY per advance,

Ronflex2k.

**Comment by gioman** [comment]
> Hello, I'm very interested by this feature but I don't know how implement this thing in QGis. Do you have some advices ?

@Ronflex2k this functionality has already been implemented.

**Comment by Ronflex2k** [comment]
> > Hello, I'm very interested by this feature but I don't know how implement this thing in QGis. Do you have some advices ?
> 
> @Ronflex2k this functionality has already been implemented.

I run QGis 3.16.16 LTR and didn't appears. Should I run 3.22 ?

**Pull request merged**